apis/components/component.go

```diff
@@ -0,0 +1,68 @@
+// +groupName=datasciencecluster.opendatahub.io
```

Comment:
I notice this file doesn't have a license header like all (or at least most?) of the other Go files added here, is that intentional?

Also, should the year on those headers be 2024 rather than 2023?

Reply:
Those are operator-sdk generated headers. Since component.go file was create by us, it did not include the header. We currently do not have a uniform way to apply license to individual files across `opendatahub-io` repositories. Maybe something we can bring up to get inputs from all repos?

Reply:
Ah right, that makes sense :+1: It's several years since I used the operator-sdk or Go, so please excuse my rustiness :sweat_smile: 

I wouldn't bother adding the license header to all files then. From what I remember, that's not actually important really.